### PR TITLE
Fix whitespace appearance in CTA links with icons.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #7968

## Relevant technical choices

Changed the link to be a `display: inline-block;` with a `position: relative;` icon that is manually aligned instead of flex aligned.

Seems to work even with multi-line links:

CleanShot 2023-12-08 at 17 22 52

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @tofumatt – this is almost there, just a few comments.

Also, [this VRT](https://github.com/google/site-kit-wp/pull/7969/files#diff-8bfa23056b016d3bcbb34b140576dfd0a42c6c361c86076b29a455feb77bb7a2) looks quite off but maybe it's just a fluke as I don't see it that way in the browser.
image

File: assets/js/components/Link.js [hunk 1]
```diff
@@ -178,18 +178,18 @@ const Link = forwardRef( ( props, ref ) => {
 			{ ...otherProps }
 		>
 			{ back && (
-				<IconWrapper marginRight={ 5 }>
```
[the remark this answers is not in the document]
Comment (tofumatt) [reply assets/js/components/Link.js:181]: The margin left and right props were there before, and I _think_ still needed.

[the remark this answers is not in the document]
Comment (tofumatt) [reply assets/js/components/Link.js:181]: Oh, wait, yes, the `top` isn't needed.

File: assets/js/components/adminbar/__snapshots__/AdminBarWidgets.test.js.snap [hunk 2]
```diff
@@ -98,80 +98,24 @@ exports[`AdminBarWidgets should render the Admin Bar Widgets, including the Acti
       class="mdc-layout-grid__cell mdc-layout-grid__cell--span-3-desktop mdc-layout-grid__cell--span-2-tablet"
     >
       <div
-        class="googlesitekit-data-block overview-total-impressions googlesitekit-data-block--default"
-        tabindex="-1"
+        class="googlesitekit-preview-block"
+        style="width: auto; height: 59px;"
```
Comment (aaemnnosttv) [inline assets/js/components/adminbar/__snapshots__/AdminBarWidgets.test.js.snap:102]: This snapshot test is failing https://github.com/google/site-kit-wp/actions/runs/7197381611/job/19604375682?pr=7969#step:7:465

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @tofumatt – looks like one last thing to resolve

File: assets/js/components/adminbar/__snapshots__/AdminBarWidgets.test.js.snap (hunk 2 above)
Comment (tofumatt) [reply assets/js/components/adminbar/__snapshots__/AdminBarWidgets.test.js.snap:102]: Oh, weird, I ran the update. I'll do it again 😅

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks @tofumatt !

**Pull request merged**